studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx

```diff
@@ -20,13 +21,14 @@ const GridHeaderActions: FC<Props> = ({
   setApiPreviewPanelOpen,
   refreshDocs,
 }) => {
-  const { ui, meta } = useStore()
-  const projectRef = ui.selectedProject?.ref
+  const { meta } = useStore()
+  const { project } = useProjectContext()
+  const projectRef = project?.ref
   const policies = meta.policies.list((policy: PostgresPolicy) => policy.table_id === table.id)
 
-  const isReadOnly =
-    !checkPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'tables') &&
-    !checkPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'columns')
+  const canSqlWriteTables = useCheckPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'tables')
+  const canSqlWriteColumns = useCheckPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'columns')
+  const isReadOnly = canSqlWriteTables && canSqlWriteColumns
```

Comment:
it should be `!` right?

```suggestion
  const isReadOnly = !canSqlWriteTables && !canSqlWriteColumns
```